assets/js-text-with-image.js

```diff
@@ -3,79 +3,168 @@
  *
  * Ensures that spacer elements in text-with-image sections
  * inherit the correct color palette from the previous section.
+ * Optimized for performance with Utils library functions.
+ *
+ * @requires js-utils-core.js
+ * @requires js-utils.js
  */
 const handleTextWithImageSpacer = (section) => {
   if (!section) return null;
 
   const config = {
-    classes: {
+    class: {
       spacer: 'text-image__spacer',
       palettes: ['palette-one', 'palette-two', 'palette-three']
-    },
-    selectors: {
-      spacer: '.text-image__spacer'
     }
   }
 
-  const elements = {
-    children: [...section.children],
-    prevSection: section.previousElementSibling
-  }
+  // Cache elements but get previous section on demand
+  const getElements = () => ({
+    section,
+    prevSection: $.getSibling(section, '*', 'prev')
+  })
+
+  // Cache checking functions for better performance
+  const hasPalette = (el, palette) => el.classList.contains(palette);
+  const hasSpacerClass = (el) => el.classList.contains(config.class.spacer);
 
   const applyPaletteToSpacer = (spacer, prevChild) => {
-    const prevChildPalettes = config.classes.palettes.filter(
-      paletteClass => prevChild.classList.contains(paletteClass)
-    )
-
-    const currentSpacerPalettes = config.classes.palettes.filter(
-      paletteClass => spacer.classList.contains(paletteClass)
-    )
-
-    // Apply palette classes from the previous section to the spacer
-    prevChildPalettes.forEach(palette => {
-      const currentPalette = currentSpacerPalettes[0]; // Take the first palette if multiple exist
-
-      if (currentPalette && currentPalette !== palette) {
-        // Replace the current palette with the one from the previous section
-        spacer.classList.replace(currentPalette, palette);
-      } else if (!currentPalette) {
+    let foundPrevPalette = null,
+        currentSpacerPalette = null;
+
+    const findPalette = (el) => {
+      for (let i = 0; i < config.class.palettes.length; i++) {
+        const palette = config.class.palettes[i];
+
+        if (hasPalette(el, palette)) return palette;
+      }
+      return null;
+    }
+
+    foundPrevPalette = findPalette(prevChild); // Find palettes in previous child and current spacer
+
+    if (!foundPrevPalette) return;
+
+    currentSpacerPalette = findPalette(spacer); // Find the current palette in the spacer
+
+    // Apply palette class efficiently using Utils with frame sequencing
+    const applyClasses = () => {
+      if (currentSpacerPalette && currentSpacerPalette !== foundPrevPalette) {
+        // Use frameSequence for better performance and transition handling
+        $.frameSequence(
```

Comment:
Where is this function and batchDom defined? 

I saw they were added here https://github.com/booqable/chameleon-theme/pull/281 but I'd like to see the definition to better understand them 🤔